nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -374,6 +376,217 @@ in {
         '';
       };
 
+      defaultLocale = mkOption {
+        default = null;
+        type = types.nullOr types.str;
+        example = "de_CH";
+        description = ''
+          Default locale to use when Nextcloud fails to detect the user's
+          locale.
+
+          Further details about this setting and supported locale codes can be found in the
+          <link xlink:href="https://docs.nextcloud.com/server/stable/admin_manual/configuration_server/language_configuration.html">upstream documentation</link>.
+        '';
+      };
+
+      twofactorEnforcement = let
+        twoFactorAuthUpstreamManualUrl = "https://docs.nextcloud.com/server/latest/admin_manual/configuration_user/two_factor-auth.html";
+      in {
+        enable = mkOption {
```

Comment:
This one is somewhat dangerous: when enabling this straight from the beginning, `root` cannot authenticate because of the 2fa enforcement. This should be reflected here.